src/VisualStudio/Core/Def/Implementation/ProjectSystem/DocumentProvider.cs

```diff
@@ -12,11 +12,6 @@ namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem
     [Obsolete("This is a compatibility shim for TypeScript and Live Share; please do not use it.")]
     internal sealed class DocumentProvider
     {
-        [Obsolete("This is a compatibility shim for Live Share; please do not use it.")]
-        public DocumentProvider(VisualStudioProjectTracker projectTracker, IServiceProvider serviceProvider, VisualStudioDocumentTrackingService documentTrackingService)
-        {
-        }
```

Comment:
Isn't this a problem?  How does thisnot break LiveShare?

Reply:
Yes, we've already reached out to them. This was a hack we just put in for them, so we'll pull it back out. Unfortunately maintaining this hack isn't going to work.